src/openrct2/ride/Ride.cpp

```diff
@@ -7372,6 +7372,25 @@ void Ride::IncreaseNumShelteredSections()
     num_sheltered_sections |= newNumShelteredSections;
 }
 
+bool Ride::HasRecolourableShopItems() const
+{
+    bool HasRecolourableShopItems = false;
+
+    const auto rideEntry = GetRideEntry();
+    if (rideEntry == nullptr)
+        return false;
+
+    for (int itemIndex = 0; itemIndex < NUM_SHOP_ITEMS_PER_RIDE; itemIndex++)
+    {
+        const ShopItem currentItem = rideEntry->shop_item[itemIndex];
+        if (currentItem != ShopItem::None)
+        {
+            HasRecolourableShopItems |= GetShopItemDescriptor(currentItem).IsRecolourable();
```

Comment:
You can simply return true here if the item is recolourable and return false on line 7391. Might save a few cycles if shops have multiple items to sell.